navigation/src/androidMain/kotlin/com/freeletics/khonshu/navigation/Navigator.kt

```diff
@@ -39,20 +29,30 @@ public interface Navigator {
     public fun <T : BaseRoute> navigateBackTo(popUpTo: KClass<T>, inclusive: Boolean = false)
 
     /**
-     * Reset the back stack to the given [root]. The current back stack will cleared and if
-     * root was already on it it will be recreated.
+     * Show the existing back stack for [root] or create a new back stack if none exists. If an existing back stack
+     * is shown it will be shown in it's previous state. If [root] is the root of the current back stack this is a
+     * no-op.
+     *
+     * The back stack that is shown at the time of calling this method will not back stack will not be modified.
+     */
+    public fun switchBackStack(root: NavRoot)
+
+    /**
+     * Show [root] as the current destination. If there already is a back stack for [root], regardless of whether
+     * it is the current back stack or not, the stack will be cleared.
+     *
+     * The back stack that is shown at the time of calling this method, if it isn't the back stack of [root],
+     * will not back stack will not be modified.
```

Comment:
I'm not 100% sure about this part. It matches `resetToRoot(root)` when `root` is the current stack and it also matches `navigateToRoot(root, restoreRootState = false)` when `root` isn't current. However `resetToRoot(root)` where `root` is not the current stack but the start stack would clear both current and the start stack before. If we do change this to also clear the current stack, it would not be completely equivalent to `navigateToRoot(root, restoreRootState = false)` for the case that `root` is the start destination.

For how we use the existing methods in our code base, clearing the current stack would be more fitting. We do have multiple cases where we call `resetToRoot(StartRoot)` and expect the current stack to be cleared (usually at the end of a flow). We have 0 cases of `navigateToRoot(root, restoreRootState = false)` so it not being entirely equivalent doesn't matter to existing code.

Despite that, I think the way it's implemented in this PR and written in the doc here is better and we should probably just update the cases that we have.